### PR TITLE
EWPP-2269: Add Remote translation provider plugin type.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
 namespace Drupal\Tests\oe_translation_remote\Functional;
 
 use Drupal\node\Entity\Node;
-use Drupal\Tests\oe_translation\Functional\TranslationTestBase;
+use Drupal\Tests\oe_translation\FunctionalJavascript\TranslationTestBase;
```
Comment (upchuk) [inline modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php:8]: You made this a functional JS test but left it in the Functional namespace?

File: modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php [hunk 2]
```diff
@@ -36,4 +37,66 @@ public function testRemoteTranslationRoute(): void {
...
+    $this->assertSession()->elementAttributeContains('css', 'select#edit-plugin', 'required', 'required');
+
+    // Assert different configuration form based on selected plugin.
+    $this->getSession()->getPage()->selectFieldOption('Plugin', 'Remote one');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Plugin configuration for Remote one');
+    $this->assertSession()->pageTextContains('This plugin does not have any configuration options.');
+    $this->getSession()->getPage()->selectFieldOption('Plugin', 'Remote two');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Plugin configuration for Remote two');
```
Comment (upchuk) [inline modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php:76]: Assert the page no longer contains the config form for remote one.

File: modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php [hunk 3]
```diff
@@ -36,4 +37,66 @@ public function testRemoteTranslationRoute(): void {
...
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->assertSession()->pageTextContains('Created the Test provider Remote Translator Provider.');
+
+    // Edit the provider.
+    $this->clickLink('Edit');
+    $this->assertSession()->fieldValueEquals('Name', 'Test provider');
+    $this->assertSession()->fieldValueEquals('A test configuration string', 'Plugin configuration.');
+    $this->getSession()->getPage()->fillField('Name', 'Test provider edited');
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->assertSession()->pageTextContains('Saved the Test provider edited Remote Translator Provider.');
```
Comment (upchuk) [inline modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php:93]: Do also a load of the config entity and assert all the exported values are in the correct format.

File: tests/src/Functional/TranslationTestBase.php [hunk 4]
```diff
@@ -65,7 +65,6 @@ protected function setUp(): void {
     $role = $this->entityTypeManager->getStorage('user_role')->load('oe_translator');
     $permissions = $role->getPermissions();
     $permissions[] = 'administer menu';
-    $permissions[] = 'edit any page content';
```
Comment (upchuk) [inline tests/src/Functional/TranslationTestBase.php:68]: Why did you remove this?

File: tests/src/FunctionalJavascript/TranslationTestBase.php [hunk 5]
```diff
@@ -53,10 +58,16 @@ protected function setUp() {
...
+    $this->drupalPlaceBlock('page_title_block', ['region' => 'content']);
+    $this->drupalPlaceBlock('local_tasks_block', ['region' => 'content']);
+    $this->drupalPlaceBlock('local_actions_block', ['region' => 'content']);
+
     /** @var \Drupal\user\RoleInterface $role */
     $role = $this->entityTypeManager->getStorage('user_role')->load('oe_translator');
     $permissions = $role->getPermissions();
     $permissions[] = 'administer menu';
+    $permissions[] = 'edit any page content';
+    $permissions[] = 'administer site configuration';
```
Comment (upchuk) [inline tests/src/FunctionalJavascript/TranslationTestBase.php:70]: Well, we cannot add this permission that essentially allows them to do anything. It's no longer a translator then. You need to keep this user a regular translator, which may create some content, fine, but not do everything. Make another user where you need the proper role for creating provider entities.

File: modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php [hunk 6]
```diff
@@ -36,4 +37,66 @@ public function testRemoteTranslationRoute(): void {
     $this->assertSession()->pageTextContains('The remote translations overview page.');
   }
 
+  /**
+   * Test the remote translation provider configuration.
+   */
+  public function testRemoteTranslationProvider(): void {
```
Comment (upchuk) [inline modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php:43]: Name the tests a bit better, this is very vague and the test doesn't do this. The test tests that creation of a translation provider config. So something like `testRemoteTranslationProviderForm` is better.

File: tests/src/Functional/TranslationTestBase.php (hunk 4 above)
Comment (22Alexandra) [reply tests/src/Functional/TranslationTestBase.php:68]: Because I previously added it as it was needed for `testRemoteTranslationRoute` but now that I switched to functional JS, I moved the permission in that base class. The permission itself is only temporary needed anyway as it's supposed to be replaced later.

**Review by upchuk — DISMISSED** [review]

**Comment by hernani** [comment]
image
Only problem found relates to dashboard being visible in view tab . Also translate/view tab seem both active.

**Comment by upchuk** [comment]
@hernani Like I said, this is now fixed with the merge of the local translations.

**Review by upchuk — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]
